storage/Devices/LvmVg.h

```diff
@@ -54,9 +63,43 @@ namespace storage
 	unsigned long long get_size() const;
 
 	/**
+	 * Returns the extent size of the volume group.
+	 *
+	 * @return extent size
+	 */
+	unsigned long long get_extent_size() const;
+
+	/**
+	 * Set the extent size of the volume group. This can modify the size
+	 * of the logical volumes.
+	 *
+	 * @throw InvalidExtentSize
+	 */
+	void set_extent_size(unsigned long long extent_size);
+
+	/**
+	 * Adds a block device as a physical volume to the volume group. If
+	 * there is not a physical volume on the block device it will be
+	 * created.
+	 */
+	LvmPv* add_pv(BlkDevice* blk_device);
+
+	/**
+	 * Removes a block device from the volume group. The physical volume
+	 * on the block device will the deleted.
+	 */
+	void remove_pv(BlkDevice* blk_device);
+
+	/**
 	 * Create a logical volume with name lv_name in the volume group.
 	 */
-	LvmLv* create_lvm_lv(const std::string& lv_name);
+	LvmLv* create_lvm_lv(const std::string& lv_name, unsigned long long size);
```

Comment:
This is somewhat inconsistent: add_pv()/remove_pv() vs. create_lvm_lv()/delete_lvm_lv().

Since you need to create an instance of this LvmVg class anyway to do this, how about omitting that redundant _lvm in those calls?